Jenkinsfile

```diff
@@ -1,34 +1,64 @@
 #!/usr/bin/env groovy
 
-/**
-* Checks if as message contains a given command and machine.
-*
-* @param message The actual message.
-* @param command The command.
-* @param machine The machine.
-* @return A boolean indicating whether there is a match.
-*/
-boolean machineCheck(String message, String command, String machine) {
-    def matchPattern = "@jenkins-cscs\\s+${command}\\s+${machine}"
-    return message ==~ machinePattern
-}
-
 def dirPrefix = 'reframe-ci'
 def loginBash = '#!/bin/bash -l'
 def bashScript = 'ci-scripts/ci-runner.bash'
 def cscsSettings = 'config/cscs.py'
 def machinesList = ['daint', 'dom', 'kesch', 'leone', 'monch']
 def githubComment = env.ghprbCommentBody
 def machinesToRun = machinesList
-def shortCommit
+def uniqueID
 
 stage('Initialization') {
     node('master') {
         try {
             def scmVars = checkout scm
-            shortCommit = scmVars.GIT_COMMIT[0..6]
+            uniqueID = "${scmVars.GIT_COMMIT[0..6]}${env.BUILD_ID}"
```

Comment:
I would put a hyphen between the commit hash and the build ID, so as not to be confused as a single number.